api-service/service/hosts.go

```diff
@@ -49,41 +53,66 @@ func (as *APIService) SearchHostsAsLMS(filters dto.SearchHostsFilters) (*exceliz
 		return nil, aerr
 	}
 
+	var j int
 	for i, val := range hosts {
 		i += 4 // offset for headers
-		lms.SetCellValue("Database_&_EBS_DB_Tier", fmt.Sprintf("B%d", i), val["physicalServerName"])
-		lms.SetCellValue("Database_&_EBS_DB_Tier", fmt.Sprintf("C%d", i), val["virtualServerName"])
-		lms.SetCellValue("Database_&_EBS_DB_Tier", fmt.Sprintf("D%d", i), val["virtualizationTechnology"])
-		lms.SetCellValue("Database_&_EBS_DB_Tier", fmt.Sprintf("E%d", i), val["dbInstanceName"])
-		lms.SetCellValue("Database_&_EBS_DB_Tier", fmt.Sprintf("F%d", i), val["pluggableDatabaseName"])
-		lms.SetCellValue("Database_&_EBS_DB_Tier", fmt.Sprintf("G%d", i), val["environment"])
-		lms.SetCellValue("Database_&_EBS_DB_Tier", fmt.Sprintf("H%d", i), val["options"])
-		lms.SetCellValue("Database_&_EBS_DB_Tier", fmt.Sprintf("I%d", i), val["usedManagementPacks"])
-		lms.SetCellValue("Database_&_EBS_DB_Tier", fmt.Sprintf("N%d", i), val["productVersion"])
-		lms.SetCellValue("Database_&_EBS_DB_Tier", fmt.Sprintf("O%d", i), val["productLicenseAllocated"])
-		lms.SetCellValue("Database_&_EBS_DB_Tier", fmt.Sprintf("P%d", i), val["licenseMetricAllocated"])
-		lms.SetCellValue("Database_&_EBS_DB_Tier", fmt.Sprintf("Q%d", i), val["usingLicenseCount"])
-
-		hostname := val["physicalServerName"].(string)
-		if len(hostname) == 0 {
-			hostname = val["virtualServerName"].(string)
-		}
-		if csi, ok := csiByHostname[hostname]; ok {
-			lms.SetCellValue("Database_&_EBS_DB_Tier", fmt.Sprintf("R%d", i), strings.Join(csi, ", "))
+		setCellValueLMS(lms, sheetDatabaseEbsDbTier, i, csiByHostname, val)
+		createdDate := val["createdAt"].(primitive.DateTime).Time().UTC()
+		if (filters.From != utils.MIN_TIME ||
+			filters.To != utils.MAX_TIME) &&
+			createdDate.After(filters.From) &&
+			createdDate.Before(filters.To) {
+			if j == 0 {
+				j = 4 // offset for headers
```

Comment:
If the first `val` is not an and host that belong to `Host added` sheet, we do never enter in this if and we never adjust the headers offset for the `j` variable!
We must set the offset out of the `range hosts` loop.

Reply:
I just understood this is correct and working :)

Reply:
Ok, thank you!